suite/tests/linux/clone.c

```diff
@@ -124,20 +129,23 @@ main()
     int ret_failure_clone3 = make_clone3_syscall(NULL, 0, NULL);
     assert(ret_failure_clone3 == -1);
 
-#ifndef ANDROID /* clone3 not supported on Android */
+    /* On some environments, we see that the kernel supports clone3 even though
+     * SYS_clone3 is not defined by glibc.
```

Comment:
This doesn't all seem consistent: it makes a clone3 syscall outside of any ifdefs, and has a comment saying clone3 may be available even if SYS_clone3 is not defined, yet it puts the enabling and use of it back under an ifdef for SYS_clone3?  It doesn't seem worth trying the syscall, or implying here we should go ahead and use clone3, if below we go back to the ifdef for what matters, setting clone3_available.  This code sequence lines 128 through 147 would be simpler and more understandable if it were all under ifdef SYS_clone3 if we're never going to use clone3 if that's not defined.

Reply:
One needs to go back to how things were, pre- #6597.
That code was calling clone3 outside of ifdefs, presumably to make sure the right errno was returned?
Removing that is outside the scope of #6596 (which #6597 was an attempt to fix).

Reply:
I don't follow the argument: this PR is taking a sequence of code that does work to compute `clone3_available` and adds an ifdef only around the final step instead of around all the work, when all the work is *only* for computing that final step `clone3_available`: so it is this PR which is introducing the awkward code.  Not seeing why the history matters.

Reply:
> Question: Why does it make a clone3 syscall ifndef SYS_clone3 ?
> That code was calling clone3 outside of ifdefs, presumably to make sure the right errno was returned?

Maybe your question is why do we need to verify clone3 behaviour if SYS_clone3 is not defined?

Note that clone3 is one of the syscalls that DR intercepts. So if SYS_clone3 is not defined but the app still tries to make the syscall, we want to make sure that DR returns the correct error code.

Reply:
Rewrote patch to use DR's copy of struct clone_args from core/unix/include/clone3.h.